model/task_queue_service_dependency.go

```diff
@@ -354,6 +345,27 @@ func (t *basicCachedDAGDispatcherImpl) nextTaskGroupTask(unit schedulableUnit) *
 			return nil
 		}
 
+		// Check if its dependencies have been met.
+		depsMet, err := nextTaskFromDB.DependenciesMet(make(map[string]task.Task))
```

Comment:
what's the point of passing an empty but constructed map to this function? is it a mechanism that we should be using to cache? should we make a variant of this method that doesn't take an argument? Something smells wrong about this.

Reply:
As discussed in person: will pass reference.